src/parser/monk/mistweaver/modules/spells/essences/WayOfTheCrane.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+import StatisticBox from 'interface/others/StatisticBox';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+
+
+
+/**
+ * Simple info graphic that shows what spell did what healing during wotc window, the hps of each spell, and hptc
+ */
+
+class WayOfTheCrane extends Analyzer {
+
+  customMap = null;
+  damageSpell = "";
+  lastTimeStamp = 0;
+  inWotc = false;
+  wotcTime = 0;
+
+  constructor(...args) {
+    super(...args);
+    const hasEssence = this.selectedCombatant.hasEssence(SPELLS.CONFLICT.traitId);
+    if(!hasEssence){
+        return;
+    }
+    this.active = this.selectedCombatant.hasMajor(SPELLS.CONFLICT.traitId);
+    if (!this.active) {
+      return;
+    }
+    this.customMap = new Map();
+  }
+
+  on_byPlayer_damage(event) {
+    if(!this.selectedCombatant.hasBuff(SPELLS.WAY_OF_THE_CRANE.id)){
```

Comment:
if you use an eventfilter instead of the old event methods you can avoid having to check for IDs in here which to me seems cleaner

Reply:
I'm not sold on eventfilters. Not my cup O'tea